marketplacetests/tests/test_marketplace_incorrect_pin.py

```diff
@@ -0,0 +1,44 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from fxapom.fxapom import FxATestAccount
+
+from marketplacetests.marketplace_gaia_test import MarketplaceGaiaTestCase
+from marketplacetests.marketplace.app import Marketplace
+from marketplacetests.payment.app import Payment
+
+
+class TestMarketplaceIncorrectPin(MarketplaceGaiaTestCase):
+
+    def test_incorrect_pin(self):
+
+        APP_NAME = 'Test Zippy With Me'
+        good_pin = '1234'
+        bad_pin = '1111'
```

Comment:
The one advantage I like is the ability to log all test data in one go with the string representation of a dict. Also, it can be handy for injecting test data, or to use a list of dictionaries for parallelized tests.